src/main/java/net/sf/jabref/shared/DBMSConnectionProperties.java

```diff
@@ -75,4 +91,33 @@ public void setPassword(String password) {
         this.password = password;
     }
 
+    /**
+     *  Gets all required data from {@link SharedDatabasePreferences} and sets them if present.
+     */
+    private void setFromPreferences(SharedDatabasePreferences prefs) {
+        if (prefs.getType().isPresent()) {
+            Optional<DBMSType> dbmsType = DBMSType.fromString(prefs.getType().get());
+            if (dbmsType.isPresent()) {
+                this.type = dbmsType.get();
+            }
+        }
+
+        prefs.getHost().ifPresent(theHost -> this.host = theHost);
+        prefs.getPort().ifPresent(thePort -> this.port = Integer.parseInt(thePort));
+        prefs.getName().ifPresent(theDatabase -> this.database = theDatabase);
+        prefs.getUser().ifPresent(theUser -> this.user = theUser);
+
+        if (prefs.getPassword().isPresent() && prefs.getUser().isPresent()) {
+            try {
+                this.password = new Password(prefs.getPassword().get().toCharArray(), prefs.getUser().get()).decrypt();
+            } catch (UnsupportedEncodingException | GeneralSecurityException e) {
+                LOGGER.error("Could not decrypt password", e);
+            }
+        }
+
+        if (!prefs.getPassword().isPresent()) {
```

Comment:
you could combine this if and the if statement above with an if / else if like the following:
```
if(!prefs.getPassword().isPresent()) {
    ...
} else if(prefs.getPassword().isPresent() && prefs.getUser().isPresent()){
  ....
}
```